src/components/anti-everyone.ts

```diff
@@ -1,16 +1,28 @@
 import * as Discord from "discord.js";
 import { BotComponent } from "../bot-component.js";
 import { Wheatley } from "../wheatley.js";
-import { colors } from "../common.js";
+import { colors, MINUTE, HOUR } from "../common.js";
 import { M } from "../utils/debugging-and-logging.js";
+import { SelfClearingMap } from "../utils/containers.js";
 
 const failed_everyone_re = /(?:@everyone|@here)/g; // todo: word boundaries?
 
+export interface AntiEveryoneMessageCache {
+    reply_to: Discord.Message["id"];
+    reply: Discord.Message;
+}
+
 /**
  * Responds to users attempting to ping @everyone or @here
  * with a message discouraging the behavior.
  */
 export default class AntiEveryone extends BotComponent {
+    /**
+     * Replies that have been made to users who attempted to ping everyone.
+     *
+     * @note This is limited to the last hour, in order to keep memory usage down.
+     */
+    public replies: SelfClearingMap<Discord.User, AntiEveryoneMessageCache[]> = new SelfClearingMap(HOUR, MINUTE);
```

Comment:
super minor comment: `public`/`private` isn't really used in the code base at the moment so for consistency it'd make sense to leave off the "public" (and also it doesn't need to be public)